common/test/acceptance/tests/helpers.py

```diff
@@ -298,6 +301,29 @@ def assert_event_emitted_num_times(self, event_name, event_time, event_user_id,
             ).count(), num_times_emitted
         )
 
+    def verify_client_side_events(self, event_type, expected_events):
+        """Verify that the expected client-side events were logged.
+
+        Args:
+            event_type (str): The type of event to be verified.
+            expected_events (list): A list of dicts representing the events that should
+                have been fired.
+        """
+        # Wait for a couple of seconds to ensure that the event was posted from the client
+        time.sleep(2)
```

Comment:
Thanks, that's a great suggestion. I don't know why I hadn't thought about doing it as a promise.